operator/slotticker/slotticker.go

```diff
@@ -36,34 +46,38 @@ func New(logger *zap.Logger, cfg Config) *slotTicker {
 }
 
 func newWithCustomTimer(logger *zap.Logger, cfg Config, timerProvider TimerProvider) *slotTicker {
-	now := time.Now()
-	timeSinceGenesis := now.Sub(cfg.GenesisTime)
+	timeSinceGenesis := time.Since(cfg.GenesisTime)
 
-	var initialDelay time.Duration
+	var (
+		initialDelay time.Duration
+		initialSlot  phase0.Slot
+	)
 	if timeSinceGenesis < 0 {
 		// Genesis time is in the future
 		initialDelay = -timeSinceGenesis // Wait until the genesis time
+		initialSlot = phase0.Slot(0)     // Start at slot 0
 	} else {
 		slotsSinceGenesis := timeSinceGenesis / cfg.SlotDuration
 		nextSlotStartTime := cfg.GenesisTime.Add((slotsSinceGenesis + 1) * cfg.SlotDuration)
 		initialDelay = time.Until(nextSlotStartTime)
+		initialSlot = phase0.Slot(slotsSinceGenesis)
 	}
 
 	return &slotTicker{
 		logger:       logger,
 		timer:        timerProvider(initialDelay),
 		slotDuration: cfg.SlotDuration,
 		genesisTime:  cfg.GenesisTime,
-		slot:         0,
+		slot:         initialSlot,
 	}
 }
```

Comment:
Seems like here we should start at some slot that corresponds to `cfg.GenesisTime` instead of at slot 0 ?

Reply:
> I'm guessing this didn't cause any issues because we always called `Next` before calling `Slot` on brand-new `slotTicker`

note btw, this ^ also means we never actually "tick" (handle duties) for the very first `initialSlot` - but rather wait until the next slot that comes after it,

at the worst case we essentially skip 1 slot during start, not a big deal probably - but maybe now we can address this, wdyt ?